### PR TITLE
Fix encap port configuration for remote chassis

### PR DESCRIPTION
**Pull request by jcaamano**

### CODE REVIEW

**Comment by jcaamano** [comment]
PTAL @trozet @tssurya

**Comment by tssurya** [comment]
nice catch @jcaamano ; adding to review list

**Comment by qinqon** [comment]
Downstream hypershift kubevirt job is all green, https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ovn-kubernetes/1831/pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-kubevirt/1694385980821213184

**Review by trozet — APPROVED** [review]

File: go-controller/pkg/ovn/zone_interconnect/chassis_handler.go [hunk 1]
```diff
@@ -152,5 +153,10 @@ func (zch *ZoneChassisHandler) createOrUpdateNodeChassis(node *corev1.Node, isRe
 		Options:     map[string]string{"csum": "true"},
 	}
 
+	// set the geneve port if using something else than default
+	if config.Default.EncapPort != config.DefaultEncapPort {
```
Comment (tssurya) [inline go-controller/pkg/ovn/zone_interconnect/chassis_handler.go:157]: do we need a log here to confirm this step is done? better debugging i guess, but if this is notified elsewhere differently I am ok not doing this here.

File: contrib/kind.sh [hunk 2]
```diff
@@ -204,6 +206,9 @@ parse_args() {
                                                 ;;
             -df | --disable-forwarding )        OVN_DISABLE_FORWARDING=true
                                                 ;;
+            -ecp | --encap-port )               shift
```
Comment (tssurya) [inline contrib/kind.sh:209]: do we want to support this as u/s user facing CLI? we should update the kind.md docs with the new option

**Review by tssurya — COMMENTED** [review]
two nits, otherwise lgtm.

File: contrib/kind.sh (hunk 2 above)
Comment (jcaamano) [reply contrib/kind.sh:209]: I think that documenting the options in kind.md via just printing out the same thing that `kind.sh --help` prints out is just a waste of time. It is already outdated and honestly a nuisance to keep in sync. I think we should just refer there to the possibility of using `kind.sh --help` where most options are self explanatory and just provide some deeper explanation on more complex configuration options. I can clean that up in a follow up PR. WDYT?

**Review by trozet — APPROVED** [review]

File: contrib/kind.sh (hunk 2 above)
Comment (tssurya) [reply contrib/kind.sh:209]: ack i am ok with deleting that doc if that's what we want, but deff that wasn't stale.. i think most of them were updating it with the new options.
so +1 to either removing it OR updating it. Stale docs are yuck.

**Pull request merged**